sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/AmqpChannelProcessor.java

```diff
@@ -370,17 +366,31 @@ private void close(T channel) {
      * which removes itself from the tracking list, then propagates the notification to the wrapped subscriber.
      */
     private static final class ChannelSubscriber<T> extends Operators.MonoSubscriber<T, T> {
+
         private final AmqpChannelProcessor<T> processor;
+        // subscriberId is only needed for logging and not every subscriber is logged, so let's make it lazy.
+        private String subscriberId = null;
 
         private ChannelSubscriber(CoreSubscriber<? super T> actual, AmqpChannelProcessor<T> processor) {
             super(actual);
             this.processor = processor;
         }
 
+        void onAdd() {
+            // most subscribers never get here and will be completed immediately after they are created.
+            subscriberId = UUID.randomUUID().toString().substring(0, 6);
```

Comment:
Is it possible to correlate a subscriber to its downstream instance? (ie. ReactorReceiver A subscribed to this and was assigned subscriberId.)

Reply:
tried it in the last commit here: https://github.com/Azure/azure-sdk-for-java/pull/29738/commits/5782b42499bb19e384a1bc5caebda413ff5880da

We'd have it via `AmqpReceiveLinkProcessor` having Id and having 1:1 relationship with `ReactorReceiver`.
the id is propagated via the reactor `Context`.

I also added a similar approach to other implementations of `CoreSubscriber`, but it doesn't cover all possible cases - we subscribe a lot, and propagate ids through all of these calls would be hard.

But as far as `ChannelSubscriber` is concerned, I believe I covered receive case. 

LMK what you think and if it makes sense.